backends/gstreamer/player.rs

```diff
@@ -621,13 +629,21 @@ impl GStreamerPlayer {
             let render_sample = {
                 let render = self.render.clone();
                 let observer = self.observer.clone();
+                let weak_video_renderer = Arc::downgrade(&video_renderer);
+
                 move |sample: gst::Sample| {
                     let frame = render
                         .lock()
                         .unwrap()
                         .get_frame_from_sample(sample)
                         .map_err(|_| gst::FlowError::Error)?;
-                    video_renderer.lock().unwrap().render(frame);
+
+                    if let Some(video_renderer) = weak_video_renderer.upgrade() {
+                        video_renderer.lock().unwrap().render(frame);
+                    } else {
+                        return Err(gst::FlowError::Flushing);
+                    };
+
                     let _ = notify!(observer, PlayerEvent::VideoFrameUpdated);
```

Comment:
Should this still happen even if `video_renderer` is gone?

Reply:
Yes, no point to hold strong ref to video renderer too long, because anyway during marshalling this event the media player could be drop and message will be discarded on half-way